llvm/test/CodeGen/AArch64/vector-masked-extract.ll

```diff
@@ -0,0 +1,663 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 3
+; RUN: llc -mtriple=aarch64 -verify-machineinstrs %s -o - | FileCheck %s --check-prefixes=CHECK,NEON-FIXED
+; RUN: llc -mtriple=aarch64 -mattr=+sve -verify-machineinstrs %s -o - | FileCheck %s --check-prefixes=CHECK,SVE-FIXED
+
+define i8 @extract_last_i8(<16 x i8> %data, <16 x i1> %mask, i8 %passthru) {
```

Comment:
To simplify the tests, and more accurately reflect how they will be used can you instead generate the mask in the function?  I'm thinking something trivial like changing `%mask` to have the same type as `%data` and then just comparing that to zero.